packages/table-schema-provider/src/withTablesList.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+
+import React from 'react';
+import { wrapDisplayName } from 'recompose';
+import type { TableSchema } from '@8base/utils';
+import { TableSchemaContext } from './TableSchemaContext';
+
+
+export type TablesListProps = {
+  tablesList: TableSchema[],
+};
+
+const withTablesList = <InputProps: TablesListProps> (
+  WrappedComponent: React$ComponentType < InputProps >,
+): Class <React$Component <$Diff<InputProps, TablesListProps>>> => {
+  class WithTablesList extends React.Component<$Diff<InputProps, TablesListProps>> {
+    render() {
+      return (
+        <TableSchemaContext.Consumer>
+          { (tablesList) => (
```

Comment:
@ej9x  Maybe we can write class method instead of creating a new function on each render?

Reply:
@zouxuoz fixed